x-pack/plugins/lens/public/datasources/form_based/dimension_panel/format_selector.tsx

```diff
@@ -172,7 +173,7 @@ export function FormatSelector(props: FormatSelectorProps) {
   const { euiTheme } = useEuiTheme();
   const { selectedColumn, onChange, docLinks } = props;
   const currentFormat = isColumnFormatted(selectedColumn)
-    ? selectedColumn.params?.format
+    ? selectedColumn.params?.format || (selectedColumn.meta?.params as unknown as ValueFormatConfig)
```

Comment:
This actually is always `undefined`, and the UI never change when you change the formatting option (only the chart reflect that change)
The `updateColumnFormat` adds the `format` property to a different level in the `column` object.
I'm not sure what the intent is here, but probably we should move the `format` into the actual `meta.params` object.

In the `TextBasedLayerColumn` you haven't defined the `format` type in the `meta.params` but is at the root level. In the `GenericIndexPatternColumn` is instead within the `params` object.


I'm also wondering why we haven't aligned `GenericIndexPatternColumn | TextBasedLayerColumn` type to put the `format` in the same place, but this is a different story.

Reply:
i aligned with the formatbasedlayer and put this format under params